### ISSUE TITLE
Units for MARBL

### ISSUE THREAD

**Issue by mnlevy1981** [marbl-ecosys/MARBL#41]
Right now MARBL has adopted the `cgs` unit convention from POP, but we want to enforce `mks`. This will require conversion of intent(in) and intent(out) in the driver, and will also requirement conversion in the diagnostics.

**Comment by klindsay28** [comment]
units, in metadata, should not have elemental identifiers
e.g., we should have mol, not molC

**Comment by mnlevy1981** [comment]
In recent discussions, we have talked about allowing the GCM to specify whether to use `mks` or `cgs`. There are several things to think about in this regard:

1. Would models using `mks` require a different settings file than a `cgs` model? It would be nice if different models could all use the same input files...
2. Would MARBL use one set of units regardless of what the GCM requests, but convert input and output rather than requiring the GCM to do those conversions?

I want to put together a document that lists several possible implementations, along with the pros and cons of each... but a thought I just had was to modify the YAML schema for the settings file to include potential scale factors. E.g.
```yaml
   parm_POC_diss :
      longname : Base POC dissolution length scale
      subcategory : 4. general parameters (dissolution)
      units : cm
      datatype : real
      default_value : 100e2
```

could be

```yaml
   parm_POC_diss :
      longname : Base POC dissolution length scale
      subcategory : 4. general parameters (dissolution)
      units : m
      datatype : real
      default_value : 100
      cgs :
         units : cm
         scale_factor : 100
```

this could also extend to our use of per day instead of per second for some parameters:

```yaml
         PCref :
            longname : Maximum C-spec growth rate at Tref
            subcategory : 10. autotrophs
            units : 1/day
            datatype : real
            default_value :
               default : 1e34
               ((autotroph_sname)) == "sp" : 5
               ((autotroph_sname)) == "diat" : 5
               ((autotroph_sname)) == "diaz" : 2.5
            mks:
               units : 1/s
               scale_factor : 1/86400
            cgs:
               units : 1/s
               scale_factor : 1/86400
```

And maybe we want `reciprocal_scale_factor : 86400` or something, but the basic idea is that at some point we would define our unit system as `mks` or `cgs`; when reading the default values, if `${varname} : ${unit_system}` is in the dictionary then apply the appropriate scale factor, otherwise just use the given default value as-is. I think one potential issue with the `PCref` example is that the yaml file is used to generate a pseudo-namelist file, and we used the `_per_day` to avoid having `5.787e-5` in that text file... so maybe we would need to keep `PC_ref = 5` in the text file but also include `recip_scale_factor_for_PC_ref = 86400` and then have MARBL apply scale factors if they are included in the file?

**Comment by mnlevy1981** [comment]
In meeting yesterday, one proposal was to break this issue into smaller pieces. As a first step, we would keep the inputs and computation in `cgs` but provide a mechanism for getting diagnostics in `mks`.

@klindsay28 proposed one way to do this:

1. add a `ldiags_in_mks` parameter
2. generalize the `units` that are passed as arguments to `diags%add_diagnostic()` (e.g. `length / time` instead of `cm / s` or `m / s`)
3. introduce a function that (a) sets the correct units in `marbl_single_diagnostic_type`, and (b) stores a `scale_factor` on that type
4. introduce a function that can be called after the `compute()` routines that loops through all the diagnostics and applies the scale factor

This will ensure that the diagnostic values received by the GCM are in the correct units

**Comment by mnlevy1981** [comment]
closed by #430 (which is linked to this issue, unclear why github left it open)
